numpy/lib/histograms.py

```diff
@@ -184,10 +191,12 @@ def _hist_bin_auto(x):
     --------
     _hist_bin_fd, _hist_bin_sturges
     """
-    # There is no need to check for zero here. If ptp is, so is IQR and
-    # vice versa. Either both are zero or neither one is.
-    return min(_hist_bin_fd(x), _hist_bin_sturges(x))
-
+    fd_bw = _hist_bin_fd(x)
+    if fd_bw:
+        return min(_hist_bin_fd(x), _hist_bin_sturges(x))
```

Comment:
No point recalculating fd_bw here.

Might be tidier to compute `fd_sturges` once before both paths too.

Reply:
Oh dear, that was my initial plan, but looks like I forgot when I put the commit together haha.